cpp/include/kvikio/file_handle.hpp

```diff
@@ -84,7 +91,21 @@ inline int open_fd(const std::string& file_path,
                    mode_t mode)
 {
   // NOLINTNEXTLINE(cppcoreguidelines-pro-type-vararg)
-  int fd = ::open(file_path.c_str(), open_fd_parse_flags(flags, o_direct), mode);
+  int fd = -1;
+  if (o_direct) {
+#if defined(O_DIRECT)
+    // On Linux, use `O_DIRECT`
+    fd = ::open(file_path.c_str(), open_fd_parse_flags(flags, true), mode);
+#elif defined(F_NOCACHE)
+    // On macOS, pass `F_NOCACHE` to `fcntl` after opening file
+    fd = ::open(file_path.c_str(), open_fd_parse_flags(flags, false), mode);
+    if (fd != -1) { fcntl(fd, F_NOCACHE, 1); }
```

Comment:
Added similar logic for macOS. Would be good to check whether we need this

Reply:
Perhaps @madsbk can explain what we want `O_DIRECT` for? Are we doing something magic in this scenario that means we don't want the kernel filesystem layer handling caching?

Reply:
Yes, or rather, cuFile does some magic.

Reply:
Since cufile is not supported on macos, we therefore do not want this part on macos.

Reply:
Good point, instead of this, we could have `detail::open_fd()` fail if `O_DIRECT` isn't available and then fall back to compatibility mode: 
https://github.com/rapidsai/kvikio/blob/branch-24.08/cpp/include/kvikio/file_handle.hpp#L167-L171